lib_eio_linux/eio_linux.ml

```diff
@@ -896,6 +896,61 @@ module Low_level = struct
     Eio_unix.run_in_systhread @@ fun () ->
     Unix.getaddrinfo node service []
     |> List.filter_map to_eio_sockaddr_t
+
+    module Process = struct
+      external pidfd_open : int -> Unix.file_descr = "caml_eio_pidfd_open"
+      external pidfd_send_signal : Unix.file_descr -> int -> unit = "caml_eio_pidfd_send_signal"
+
+      external pidfd_wait : Unix.file_descr -> Unix.process_status = "caml_eio_pidfd_wait"
+
+      type t = {
+        process : FD.t;
+        mutable hook : Switch.hook;
+        mutable status : Unix.process_status Promise.t option;
+      }
+
+      let wait t = match t.status with
+        | Some p -> Promise.await p
+        | None ->
+          let p, r = Promise.create () in
+          t.status <- Some p;
+          await_readable t.process;
+          Switch.remove_hook t.hook;
+          let status = pidfd_wait (FD.to_unix `Peek t.process) in
+          Promise.resolve r status;
+          status
+
+      let resolve_program ~paths prog =
+        if not (Filename.is_implicit prog) then Some prog
+        else
+        let exists path =
+          let p = Filename.concat path prog in
+          if Sys.file_exists p then Some p else None
+        in
+        List.find_map exists paths
+      
+      let spawn ?env ?cwd ?stdin ?stdout ?stderr ~sw prog argv =
+        let paths = Option.map (fun v -> String.split_on_char ':' v) (Sys.getenv_opt "PATH") |> Option.value ~default:[ "/usr/bin"; "/usr/local/bin" ] in
+        let prog = match resolve_program ~paths prog with
+          | Some prog -> prog
+          | None -> raise (Eio.Fs.err (Eio.Fs.Not_found (Eio_unix.Unix_error (Unix.ENOENT, "", ""))))
+        in
+        let stdin = Option.map (FD.to_unix `Peek) stdin in
+        let stdout = Option.map (FD.to_unix `Peek) stdout in
+        let stderr = Option.map (FD.to_unix `Peek) stderr in
```

Comment:
Using `Peek` isn't safe (I'm planning to remove it). Use `FD.use_exn` instead to keep the FD open during the spawn operation.

Reply:
Thanks!